packages/contracts/contracts/TransactionManager.sol

```diff
@@ -336,6 +332,9 @@ contract TransactionManager is ReentrancyGuard, ProposedOwnable, ITransactionMan
 
     // First determine if this is sender side or receiver side
     if (invariantData.sendingChainId == _chainId) {
+      // Check the sender is correct
+      require(msg.sender == invariantData.initiator, "#P:039");
```

Comment:
this likely isn't feasible/useful but, is there any way we could return in the revert error the "bad initiator" here? would that be useful to know?

Reply:
just realized this is `msg.sender`, lol dumb q

Reply:
the error code maps to `INITIATOR_MISMATCH`